test/auto_parallel/hybrid_strategy/pir_reshard_nd_mesh_cross_mesh.py

```diff
@@ -0,0 +1,164 @@
+# limitations under the License.
+
+import os
+
+import paddle
+import paddle.distributed as dist
+from paddle.distributed.auto_parallel.static.pir_pass import (
+    apply_reshard_pass,
+)
+
+
+class TestReshardNdMeshCrossMesh:
+    def __init__(self):
+        self._dtype = os.getenv("dtype")
+        self._seed = eval(os.getenv("seed"))
+        self.BATCH_SIZE = 2
+        self.SEQ_LEN = 4
+        self.HIDDEN_SIZE = 8
+        self._backend = os.getenv("backend")
+        self._mesh0 = dist.ProcessMesh([[0, 1], [2, 3]], dim_names=["x", "y"])
+        self._mesh1 = dist.ProcessMesh([[4, 5], [6, 7]], dim_names=["x", "y"])
+
+    def validate(
+        self, op, tgt_operand, tgt_result, tgt_in_value, tgt_out_value
+    ):
+        # tgt_* are tuples, format: (process_ids, dims_mapping, partial_status)
+        operand_dist_attr = op.dist_attr.operand(0).as_tensor_dist_attr()
+        result_dist_attr = op.dist_attr.result(0).as_tensor_dist_attr()
+
+        assert operand_dist_attr.process_mesh.process_ids == tgt_operand[0]
+        assert operand_dist_attr.dims_mapping == tgt_operand[1]
+        assert operand_dist_attr.partial_status == tgt_operand[2]
+
+        assert result_dist_attr.process_mesh.process_ids == tgt_result[0]
+        assert result_dist_attr.dims_mapping == tgt_result[1]
+        assert result_dist_attr.partial_status == tgt_result[2]
+
+        in_value = op.operand_source(0)
+        out_value = op.result(0)
+        assert in_value.is_dist_dense_tensor_type()
+        assert out_value.is_dist_dense_tensor_type()
+        assert in_value.dist_attr().process_mesh.process_ids == tgt_in_value[0]
+        assert in_value.dist_attr().dims_mapping == tgt_in_value[1]
+        assert in_value.dist_attr().partial_status == tgt_in_value[2]
+
+        assert (
+            out_value.dist_attr().process_mesh.process_ids == tgt_out_value[0]
+        )
+        assert out_value.dist_attr().dims_mapping == tgt_out_value[1]
+        assert out_value.dist_attr().partial_status == tgt_out_value[2]
+
+    def create_program(self, input_shape, input_placements, output_placements):
+        paddle.enable_static()
+        if self._backend == "cpu":
+            paddle.set_device("cpu")
+            place = paddle.CPUPlace()
+        elif self._backend == "gpu":
+            place = paddle.CUDAPlace(dist.get_rank())
+
+        with paddle.pir_utils.IrGuard():
+            main_program = paddle.base.Program()
+            with paddle.base.program_guard(main_program):
+                input = paddle.ones(name='input', shape=input_shape)
+                dist_input = dist.shard_tensor(
+                    input, self._mesh0, input_placements
+                )
+                dist_out = paddle._C_ops.reshard(
+                    dist_input, self._mesh1, output_placements
+                )
+            dist_program = main_program.clone()
+            apply_reshard_pass(dist_program)
```

Comment:
cross mesh scenario should call apply_partition_pass first to remove operation not belong to cur rank. (fix in next PR)